src/core_modules/capture-core/metaDataStoreLoaders/programs/loadRulesCentricMetadata.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import { getContext } from '../context';
+import {
+    storeProgramRules,
+    storeProgramRulesVariables,
+    storeProgramIndicators,
+} from './quickStoreOperations';
+
+/**
+ * Load program rules, program rules variables and program indicators
+ * @param {string[]} programIds: ids for programs to load
+ */
+export const loadRulesCentricMetadata = (programIds: Array<string>) => {
+    const { storageController, storeNames } = getContext();
+
+    /**
+     * Gets the ids from IndexedDB for the specified programs
+     * (ids being program rule ids, program variable ids or program indicator ids)
+     * @param {string} storeName: name of the store
+     */
+    const getCachedIds = async (storeName): Promise<Array<string>> =>
+        // $FlowFixMe
+        (await Promise.all(
+            programIds
+                .map(programId =>
+                    storageController.getAll(
+                        storeName, {
+                            project: item => item.id,
+                            onIDBGetRequest: source => source
+                                .index('programId')
+                                .openCursor(IDBKeyRange.only(programId)),
+                        },
+                    ),
+                ),
+        ))
+            .flat(1);
+
+    /**
+     * Compares the ids that were retrieved from the api (and stored) to the ones in the cache.
+     * Removes cache records that weren't retrieved from the api
+     * @param {string[]} loadedIds: retrieved and stored ids
+     * @param {string[]} cachedIds: ids in the cache
+     * @param {string} storeName: name of the store
+     */
+    const removeUnavailableRecords = async (loadedIds, cachedIds, storeName) => {
+        const loadedIdsAsObject = loadedIds
+            .reduce((acc, ruleId) => {
+                acc[ruleId] = ruleId;
+                return acc;
+            }, {});
+
+        const unavailableIds = cachedIds
+            .filter(cachedId => !loadedIdsAsObject[cachedId]);
+
+        unavailableIds.length > 0 && await storageController.remove(storeName, unavailableIds);
+    };
+
+    /**
+     * Retrieves data from the api and stores it in IndexedDB. Removes cached programs that is no longer available.
+     * @param {string} storeName: The name of the store
+     * @param {function} storeFn: The callback function that executes the storing
+     */
+    const load = async ({ storeName, storeFn }) => {
```

Comment:
Is this naming here accurate?

Reply:
Can we try `removeUnavailableRecordsFromCache`. The function removes items from the storage, I cant see why load is the best name. 

I dont think making the code readable by using names that describe what a function does is a waste of our time. One month later when someone comes and reads this bit of code they will better understand the code if a name is chosen that describes the functionality of a method.

Reply:
> Making the code readable by using names that describe what a function does is gonna help us in the future to better reason about the code. One month later when someone comes and reads this bit of code they will better understand the code if a name is chosen that describes the functionality of a method.

Obviously I don't disagree with this 🙂 

For the actual naming here: `removeUnavailableRecordsFromCache ` is just part of what is happening here. What happens in these 3 lines: retrieve data (being program rules, program variables or program indicators) from api, save to indexedDB, retrieve what is stored in indexedDB (for the given programs) and finally compare what was retrieved from the api to what is stored in indexedDB and if there are anything in indexedDB that wasn't retrieved from the api, remove it (because this will be data the user don't have access to anymore).

I've chosen load as the function name because we're filling up the cache with the applicable data.

Reply:
> Obviously I don't disagree with this 🙂

👍  

In my other comment I mentioned that we could have the 3 lines function as anonymous in favour of creating this abstraction. This way the reader can basically see what the is happening in those 3 lines instead of having to navigate to the `load` function just to realise that `load` is basically one of the things that take place.

Wouldnt that make it more readable in your opinion?

Reply:
I disagree (obviously). But since this discussion over a minor suggestion has been going on for far too long (and I really need this PR merged), I'm changing it.